book/visual-effects.md

```diff
@@ -413,158 +421,155 @@ and macOS, and Skia in Android.
+blended with the white background. Doing the operations in a different
+order would lead to dark-orange or black text.
+
+To handle this properly, browsers apply blending not to individual
+shapes but to a tree of [*stacking contexts*][stacking-context]. Each
+stacking context is rastered into a single 2D array of pixels and then
+blended into its parent stacking context. Note that to raster a
+stacking context you first need to raster in and child stacking
+contexts, so they can be blended into the stacking context. In other
+words, rastering a web page requires a bottom-up traversal of the tree
+of stacking contexts.
 
 [stacking-context]: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
 
-With this simplication, the layout object tree has one raster group for each
-layout object, and the display list commands that raster the group are the
-ones added during the call to `paint` on that layout object.
+In real browsers, stacking contexts are formed by HTML elements with
+certain styles, up to any descendants that themselves have such
+styles. The full definition is actually quite complicated, so in this
+chapter we'll simplify by treating every layout object as a stacking
+context.
 
 ::: {.further}
-If you look at closely at the [definition][stacking-context] of stacking
-contexts in the specification, you'll see that it says the elements that
-*induce* a stacking context are exactly those that do something having to do
-with layering (`z-index`) or visual effects. With this chapter in hand, you can
-see exactly why.
-
-One case that does *not* induce a stacking context is `overflow:scroll`. Many
-browser engineers believe that was a mistake made when designing this feature
-for the web,[^also-containing-block] because it resulted in complications when
-trying making the browsers scroll smoothly and efficiently. Why, you might ask?
-One reason is that a raster group is not only a convenient match for surfaces;
-it's also a convenient way to utilize GPU acceleration for moving around those
-surfaces. Without a stacking context the browser might (depending on the web
-page structure) have to move around multiple independent surfaces with complex
-paint orders, in lockstep.
+Mostly, elements [form a stacking context][stacking-context] because
+of CSS properties that have something to do with layering (like
+`z-index`) or visual effects (like `mix-blend-mode`). On the other
+hand, the `overflow` property, which can make an element scrollable,
+does not induce a stacking context, which I think was a
+mistake.[^also-containing-block] The reason is that inside a modern
+browser, scrolling is done on the GPU by offsetting two surfaces.
+Without a stacking context the browser might (depending on the web
+page structure) have to move around multiple independent surfaces with
+complex paint orders, in lockstep, to achieve scrolling.
```

Comment:
```suggestion
complex paint orders, in lockstep, to achieve scrolling. Fixed- and sticky-positioned elements also form a stacking
context because of their interaction with scrolling.
```